src/VirtualClient/VirtualClient.Actions/HPLinpack/HPLinpackExecutor.cs

```diff
@@ -0,0 +1,408 @@
+
+        }
+
+        /// <summary>
+        /// Executes the HPL workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            DateTime startTime = DateTime.UtcNow;
+            await this.ExecuteCommandAsync("make", $"arch=Linux_GCC", this.HPLDirectory, false, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+
+            this.SetParameters();
+            await this.ConfigureDatFileAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
+
+            string results;
+            if (this.CpuArchitecture == Architecture.X64)
+            {
+                results = await this.ExecuteCommandAsync("runuser", $"-u azureuser -- mpirun --use-hwthread-cpus -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), true, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+            else
+            {
+                results = await this.ExecuteCommandAsync("runuser", $"-u azureuser -- mpirun -np {Environment.ProcessorCount} ./xhpl", this.PlatformSpecifics.Combine(this.HPLDirectory, "bin", "Linux_GCC"), true, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+            }
+
+            DateTime endTime = DateTime.UtcNow;
+            this.LogMetrics(results, startTime, endTime, telemetryContext, cancellationToken);
+        }
+
+        private void SetParameters()
+        {
+            // gives you P*Q = Environment.ProcessorCount and P <= Q and  Q-P to be the minimum possible value.
+            this.ProcessRows = 1;
+            this.ProcessColumns = Environment.ProcessorCount;
+            for (int i = 2; i <= Math.Sqrt(Environment.ProcessorCount); i++)
+            {
+                if (Environment.ProcessorCount % i == 0)
+                {
+                    int j = Environment.ProcessorCount / i;
+                    if (j - i < this.ProcessColumns - this.ProcessRows)
+                    {
+                        this.ProcessRows = i;
+                        this.ProcessColumns = j;
+                    }
+                }
+            }
+        }
+
+        private void ThrowIfPlatformIsNotSupported()
+        {
+            if (this.Platform != PlatformID.Unix)
+            {
+                throw new WorkloadException(
+                    $"The HPL workload is currently only supported on the following platform/architectures: " +
+                    $"'{PlatformSpecifics.LinuxX64}', '{PlatformSpecifics.LinuxArm64}'.",
+                    ErrorReason.PlatformNotSupported);
+            }
+        }
+
+        private async Task ConfigureMakeFileAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            string makeFilePath = this.PlatformSpecifics.Combine(this.HPLDirectory, "setup", "Make.Linux_GCC");
```

Comment:
parameterize the Make.Linux_GCC into something like string configName/makefileName. This happens a lot